aws/wafv2_helper.go

```diff
@@ -457,20 +568,116 @@ func expandWafv2RuleAction(l []interface{}) *wafv2.RuleAction {
 	action := &wafv2.RuleAction{}
 
 	if v, ok := m["allow"]; ok && len(v.([]interface{})) > 0 {
-		action.Allow = &wafv2.AllowAction{}
+		action.Allow = expandWafv2AllowAction(v.([]interface{}))
 	}
 
 	if v, ok := m["block"]; ok && len(v.([]interface{})) > 0 {
-		action.Block = &wafv2.BlockAction{}
+		action.Block = expandWafv2BlockAction(v.([]interface{}))
 	}
 
 	if v, ok := m["count"]; ok && len(v.([]interface{})) > 0 {
-		action.Count = &wafv2.CountAction{}
+		action.Count = expandWafv2CountAction(v.([]interface{}))
+	}
+
+	return action
+}
+
+func expandWafv2AllowAction(l []interface{}) *wafv2.AllowAction {
+	action := &wafv2.AllowAction{}
+	if len(l) == 0 || l[0] == nil {
+		return action
+	}
+
+	m := l[0].(map[string]interface{})
+	if v, ok := m["custom_request_handling"]; ok && len(v.([]interface{})) > 0 {
+		action.CustomRequestHandling = expandWafv2CustomRequestHandling(v.([]interface{}))
+	}
+
+	return action
+}
+
+func expandWafv2CountAction(l []interface{}) *wafv2.CountAction {
+	action := &wafv2.CountAction{}
+	if len(l) == 0 || l[0] == nil {
+		return action
+	}
+
+	m := l[0].(map[string]interface{})
+	if v, ok := m["custom_request_handling"]; ok && len(v.([]interface{})) > 0 {
+		action.CustomRequestHandling = expandWafv2CustomRequestHandling(v.([]interface{}))
 	}
 
 	return action
 }
 
+func expandWafv2BlockAction(l []interface{}) *wafv2.BlockAction {
+	action := &wafv2.BlockAction{}
+	if len(l) == 0 || l[0] == nil {
+		return action
+	}
+
+	m := l[0].(map[string]interface{})
+	if v, ok := m["custom_response"]; ok && len(v.([]interface{})) > 0 {
+		action.CustomResponse = expandWafv2CustomResponse(v.([]interface{}))
+	}
+
+	return action
+}
+
+func expandWafv2CustomResponse(l []interface{}) *wafv2.CustomResponse {
+	if len(l) == 0 || l[0] == nil {
+		return nil
+	}
+
+	m := l[0].(map[string]interface{})
+	customResponse := &wafv2.CustomResponse{}
+
+	if v, ok := m["response_code"]; ok && v.(int) > 0 {
+		customResponse.ResponseCode = aws.Int64(int64(v.(int)))
+	}
+	if v, ok := m["response_header"]; ok && len(v.(*schema.Set).List()) > 0 {
+		customResponse.ResponseHeaders = expandWafv2CustomHeaders(v.(*schema.Set).List())
+	}
+
+	return customResponse
+}
+
+func expandWafv2CustomRequestHandling(l []interface{}) *wafv2.CustomRequestHandling {
+	if len(l) == 0 || l[0] == nil {
+		return nil
+	}
+
+	m := l[0].(map[string]interface{})
+	requestHandling := &wafv2.CustomRequestHandling{}
+
+	if v, ok := m["insert_header"]; ok && len(v.(*schema.Set).List()) > 0 {
+		requestHandling.InsertHeaders = expandWafv2CustomHeaders(v.(*schema.Set).List())
+	}
```

Comment:
similar comment as above to grab the typed value from the map directly

```suggestion
	if v, ok := m["insert_header"].(*schema.Set); ok && len(v.List()) > 0 {
		requestHandling.InsertHeaders = expandWafv2CustomHeaders(v.List())
	}
```